File: backend/app/schemas/forecast.py
from pydantic import BaseModel
from typing import Optional


class ForecastResult(BaseModel):
    forecast_month: Optional[str] = None
    predicted_expenses: Optional[float] = None
    method_used: Optional[str] = None
    reliable: bool
    reliability_note: str
    mae: Optional[float] = None
    historical_months_used: int

File: frontend/src/api/forecast.ts
import api from "./client";

export interface ForecastResult {
  forecast_month: string | null;
  predicted_expenses: number | null;
  method_used: string | null;
  reliable: boolean;
  reliability_note: string;
  mae: number | null;
  historical_months_used: number;
}

export async function getForecast(): Promise<ForecastResult> {
  const response = await api.get("/forecast");
  return response.data;
}